skbio/io/_iosources.py

```diff
@@ -157,6 +158,25 @@ def get_writer(self):
         return self.file
 
 
+class NamedTemporaryFileSource(IOSource):
+    closeable = False
+
+    def can_read(self):
+        return (isinstance(self.file, tempfile._TemporaryFileWrapper) and
```

Comment:
`tempfile._TemporaryFileWrapper` is private, is there any way we can avoid relying on it?